### ISSUE TITLE
Authentication failure

### ISSUE THREAD

**Issue by mirkoperillo** [ortuman/jackal#269]
I'm testing jackal on my local computer to understand how it works, but I'm not able to have a successful authentication

## Context
- Host machine Ubuntu 20.04
- Postgresql 12 (using docker container)
- Jackal 0.62.3 (using docker image)
- jackalctl installed locally following README instruction
- Client for test `gajim` v1.1.3 (version present in the Ubuntu repo)

Test to reproduce the problem:
- Registering a user on jackal `jackalctl user add  mirko:mirko`
- Add a new account to `gajim`: ID jabber `mirko@localhost` password `mirko`

Authentication failed, this is the connection log taken from `gajim` console XML
```
!-- Outgoing ven 14 ott 2022, 15:18:01 -->
<?xml version='1.0'?>
<stream:stream xmlns="jabber:client" version="1.0" xmlns:stream="http://etherx.jabber.org/streams" to="localhost" xml:lang="it" >

<!-- Incoming ven 14 ott 2022, 15:18:01 -->
<stream xmlns="jabber:client" version="1.0" xmlns:stream="http://etherx.jabber.org/streams" from="localhost" id="0e25ee86-ae1d-453d-bbb9-a8993cda6f97" />

<!-- Incoming ven 14 ott 2022, 15:18:01 -->
<features xmlns="http://etherx.jabber.org/streams" xmlns:stream="http://etherx.jabber.org/streams" version="1.0">
<starttls xmlns="urn:ietf:params:xml:ns:xmpp-tls">
<required />
</starttls>
</features>

<!-- Outgoing ven 14 ott 2022, 15:18:01 -->
<starttls xmlns="urn:ietf:params:xml:ns:xmpp-tls"/>

<!-- Incoming ven 14 ott 2022, 15:18:01 -->
<proceed xmlns="urn:ietf:params:xml:ns:xmpp-tls" />

<!-- Outgoing ven 14 ott 2022, 15:18:01 -->
<?xml version='1.0'?>
<stream:stream xmlns="jabber:client" version="1.0" xmlns:stream="http://etherx.jabber.org/streams" to="localhost" xml:lang="it" >

<!-- Incoming ven 14 ott 2022, 15:18:01 -->
<stream xmlns="jabber:client" version="1.0" xmlns:stream="http://etherx.jabber.org/streams" from="localhost" id="4ecfa3cc-cac4-42ee-92c4-64816f79bbfb" />

<!-- Incoming ven 14 ott 2022, 15:18:01 -->
<features xmlns="http://etherx.jabber.org/streams" xmlns:stream="http://etherx.jabber.org/streams" version="1.0">
<mechanisms xmlns="urn:ietf:params:xml:ns:xmpp-sasl">
<mechanism>SCRAM-SHA-1</mechanism>
<mechanism>SCRAM-SHA-1-PLUS</mechanism>
<mechanism>SCRAM-SHA-256</mechanism>
<mechanism>SCRAM-SHA-256-PLUS</mechanism>
<mechanism>SCRAM-SHA-512</mechanism>
<mechanism>SCRAM-SHA-512-PLUS</mechanism>
<mechanism>SCRAM-SHA3-512</mechanism>
<mechanism>SCRAM-SHA3-512-PLUS</mechanism>
</mechanisms>
</features>

<!-- Outgoing ven 14 ott 2022, 15:18:01 -->
<auth xmlns="urn:ietf:params:xml:ns:xmpp-sasl" mechanism="SCRAM-SHA-1-PLUS">cD10bHMtdW5pcXVlLCxuPW1pcmtvLHI9ZTMyMDdlMGY2YWQ0ZjkwNThlOTVkMzc2ZWQ1MDRhZjU2NTIyMTg4NGE5ODU4ZDkw</auth>

<!-- Incoming ven 14 ott 2022, 15:18:01 -->
<challenge xmlns="urn:ietf:params:xml:ns:xmpp-sasl">cj1lMzIwN2UwZjZhZDRmOTA1OGU5NWQzNzZlZDUwNGFmNTY1MjIxODg0YTk4NThkOTAtYzc5ZTc3YTYtMTg2NC00NjI1LTljMzgtOWNhMmIxNWUzNDBhLHM9MHk5T09NWXVBWjVveEhiVkUwRXpLWWhld05wRFVnL2lnSXZKVGlVTms2Zz0saT0xNTAwMA==</challenge>

<!-- Outgoing ven 14 ott 2022, 15:18:01 -->
<response xmlns="urn:ietf:params:xml:ns:xmpp-sasl">Yz1jRDEwYkhNdGRXNXBjWFZsTEN6Y01MeHNQOWxFdTBpcFlxa3RQTDBsdlMvbWJGM2djNCtEVi9nbUI1YkJDQT09LHI9ZTMyMDdlMGY2YWQ0ZjkwNThlOTVkMzc2ZWQ1MDRhZjU2NTIyMTg4NGE5ODU4ZDkwLWM3OWU3N2E2LTE4NjQtNDYyNS05YzM4LTljYTJiMTVlMzQwYSxwPVE5dFJndGh2VEh6ZVpuNXBmZDZ5WGVsazlXND0=</response>

<!-- Incoming ven 14 ott 2022, 15:18:01 -->
<failure xmlns="urn:ietf:params:xml:ns:xmpp-sasl">
<not-authorized />
</failure>
```
Any suggestions from somebody who have a working jackal installation ?

Available to give more info about my tests on this.

**Comment by mirkoperillo** [comment]
@ortuman any idea or suggestion ?

**Comment by mirkoperillo** [comment]
Well, the problem was the client. The version of Gajim contained in the repo of Ubuntu 20.04 (v 1.1.4) is too old (something like 3 years old).

I installed the last version (1.5.4) through flathub and authentication works correctly.

To give more information: I tested successfully a chat between two accounts one from gajim and the other one from yaxim (an android XMPP client) and all goes right.
